### DISCUSSION TITLE
How to log information about succesfull  SSL connection

### DISCUSSION THREAD

**Discussion by al3x4kov** [encode/httpx#2552] [Q&A]
Hi!

i have this code:
`    async with httpx.AsyncClient(verify=False if not ca_bundle else ca_bundle,
                                 timeout=timeout) as client:`
 `           rsp = await client.request(
            method="GET",
            url=url,
            headers=headers
        )`
 
ca_bundle is a path like /tmp/ca_bundle
timeout = 30.0 

Everything is ok, i get response but the main question is what i should add in my code to have in log strings like:
`console.print(f"* SSL established using {version!r} / {cipher[0]!r}")`
This line from def trace from httpx/_main.py

I found that in client.request i can add parameter `extensions`, but i can't really understand what to pass in

I tried to use TRACE logging type - doesn't work for me. I see only this lines.
<img width="1223" alt="image" src="https://user-images.githubusercontent.com/22861657/213939008-207dc0cc-8113-42be-a675-cf1c9a220722.png">

**Comment by al3x4kov** [comment]
@tomchristie @florimondmanca SOS
I need to log TLS version and type of encryption (cipher) used when I connect to server via AsyncClient.

**Comment by lovelydinosaur** [comment]
Thanks. I've escalated this into a documentation issue. See https://github.com/encode/httpx/issues/2599. Are the `httpcore` docs on the `trace` extension clear enough for you to resolve this issue?

We also have some upcoming improvements to our logging in PRs https://github.com/encode/httpx/pull/2547 and https://github.com/encode/httpcore/pull/648. Those would add some information about when the SSL is established, but don't *(currently?)* include the exact information you're requesting here.

**Reply by al3x4kov** [reply]
The main problem:
1) I found how to output min and max TLS versions from transport, also i found all cipher versions which this library can use in TLS/mTLS connections.
2) If i try to use this library to create client->server requests (REST) via TLS/mTLS , i need to output all info about each call. 

Looking back on my work experience, I can confidently say that outputting information about the version of TLS used during connection establishment and the data encryption method solves problems when troubleshooting. There are situations when, with this information available in the log, the interlocutor quickly understands what is incorrectly configured on their side.

Therefore, I am asking for guidance on how I can modify the above provided source code to output information about the version of TLS used and the data encryption method for each request.
